providers/irdma/uverbs.c

```diff
@@ -1325,10 +1326,13 @@ struct ibv_qp *irdma_ucreate_qp(struct ibv_pd *pd,
 	iwvctx = container_of(pd->context, struct irdma_uvcontext,
 			      ibv_ctx.context);
 	uk_attrs = &iwvctx->uk_attrs;
+	dev_attrs = &iwvctx->dev_attrs;
 
 	if (attr->cap.max_send_sge > uk_attrs->max_hw_wq_frags ||
 	    attr->cap.max_recv_sge > uk_attrs->max_hw_wq_frags ||
-	    attr->cap.max_inline_data > uk_attrs->max_hw_inline) {
+	    attr->cap.max_inline_data > uk_attrs->max_hw_inline ||
+	    attr->cap.max_send_wr > dev_attrs->max_qp_wr ||
+	    attr->cap.max_recv_wr > dev_attrs->max_qp_wr) {
```

Comment:
Ummm...this will probably make some existing customers/app unhappy.

The max send/recv WRs are really contingent on the what the max SGE and max inline data used in this QP as the WQE's are sized based on this. See irdma_get_wqe_shift. 

So we need a combination of the incoming max send WR's and the max WQE size for this QP [shift] to determine if we will go past the device supported WQ size. This is done irdma_get_sqdepth, irdma_get_rqdepth

max_qp_wr "assumes" the largest WQE size [8 quantas or 256Bytes] and therefore is too conservative an upper bound to check the max_send/recv WRs against for this QP. We could do better if this QP used lower values of max_send_sge/max_inline [for example, it yielded a WQE size of 1 quanta or 32 bytes]. We would end up incorrectly return -EINVAL in this case.

Changing max_qp_wr to be based on the smallest WQE size also has drawbacks. i.e we would end up overflowing the WQ if the incoming  
max_send_sge/max_inline happened to make the WQE size > smallest value.

Reply:
The max values you will get out send_wr and recv_wr are 32,510 and 32767 respectively.

Reply:
@shirazsaleem : Fixed, Could you please review?

Reply:
I misunderstood your request, I thought you were trying to update an application with what to pass in max_send/recv_wr.